cli/src/pcluster/templates/cdk_builder.py

```diff
@@ -12,15 +12,151 @@
+            terminate_instance_on_failure=imagebuild.config.build_config.terminate_on_failure,
+            instance_types=["p2.xlarge", "c5.xlarge"] if archecture == "x86_64" else ["m6g.xlarge"],
+            key_pair=imagebuild.config.build_config.key_pair_name,
+        )
+
+        # Component
+        if imagebuild.config.dev_settings_config and imagebuild.config.dev_settings_config.update_and_reboot:
+            imagebuilder.CfnComponent(
+                self,
+                id="PClusterComponent",
+                name="PCluster",
+                version="0.0.1",
+                description="Bake PCluster AMI",
+                change_description="First version",
+                platform="Linux",
+                # TODO: Fix the bug using data property or uri with core.Fn.sub
+                uri="s3://test-slurm/official_ami_build_component.yml",
+            )
+        else:
+            imagebuilder.CfnComponent(
+                self,
+                id="PClusterComponent",
+                name="PCluster",
+                version="0.0.1",
+                description="Bake PCluster custom AMI",
+                change_description="First version",
+                platform="Linux",
+                # TODO: Fix the bug using data property or uri with core.Fn.sub
+                uri="s3://test-slurm/custom_ami_build_component.yml",
+            )
+
+        # ImageRecipe
+        imagebuilder.CfnImageRecipe(
+            self,
+            id="PClusterImageRecipe",
+            name="PCluster-2-10-1",
+            version="0.0.1",
+            parent_image=core.Fn.sub(
+                "arn:${AWS::Partition}:imagebuilder:${AWS::Region}:aws:image/amazon-linux-2-x86/x.x.x"
+                if not imagebuild.config.build_config.parent_image
+                else imagebuild.config.build_config.parent_image
+            ),
+            components=[
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("PClusterComponent")
+                )
+            ],
+            block_device_mappings=[
+                imagebuilder.CfnImageRecipe.InstanceBlockDeviceMappingProperty(
+                    device_name="/dev/xvda",
+                    ebs=imagebuilder.CfnImageRecipe.EbsInstanceBlockDeviceSpecificationProperty(
+                        delete_on_termination=imagebuild.config.image_config.root_volume.delete_on_termination,
+                        volume_size=imagebuild.config.image_config.root_volume.size,
+                        volume_type="gp2",
+                    )
+                    if imagebuild.config.image_config.root_volume
+                    else imagebuilder.CfnImageRecipe.EbsInstanceBlockDeviceSpecificationProperty(
+                        delete_on_termination=True, volume_size=25, volume_type="gp2"
+                    ),
+                )
+            ],
+            tags={"ParallelClusterVersion": "2.10.1"},
```

Comment:
We already have an utility function to retrieve the version of the running CLI: `utils.get_installed_version()`

Reply:
use `utils.get_installed_version()`, thanks